### ISSUE TITLE
Exception thrown when using forward only result sets

### ISSUE THREAD

**Issue by danuid** [yahoo/elide#407]
I was using SQL Server 2008 and JPA backed by Hibernate5.
The JDBC driver I'm using is `sqljdbc42.jar` from [this file](https://www.microsoft.com/en-us/download/details.aspx?id=11774).

When I try to fetch a collection using the JSON API, `ScrollableIterator.java` line 37 tried to call `this.scroll.first()` on a `ScrollableResults`.
 I'm not sure why the result set is somehow put  to `FORWARD_ONLY` mode, and an exception is thrown.

The stack trace is shown below, if it is of any help:
```
com.microsoft.sqlserver.jdbc.SQLServerException: The requested operation is not supported on forward only result sets.
	at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDriverError(SQLServerException.java:191)
	at com.microsoft.sqlserver.jdbc.SQLServerResultSet.throwNotScrollable(SQLServerResultSet.java:414)
	at com.microsoft.sqlserver.jdbc.SQLServerResultSet.verifyResultSetIsScrollable(SQLServerResultSet.java:437)
	at com.microsoft.sqlserver.jdbc.SQLServerResultSet.first(SQLServerResultSet.java:1417)
	at com.zaxxer.hikari.pool.HikariProxyResultSet.first(HikariProxyResultSet.java)
	at org.hibernate.internal.ScrollableResultsImpl.first(ScrollableResultsImpl.java:75)
	at com.yahoo.elide.datastores.hibernate5.ScrollableIterator.iterator(ScrollableIterator.java:37)
	at com.yahoo.elide.core.PersistentResourceSet.iterator(PersistentResourceSet.java:35)
	at com.yahoo.elide.core.PersistentResource.filter(PersistentResource.java:1691)
	at com.yahoo.elide.core.PersistentResource.filter(PersistentResource.java:1674)
	at com.yahoo.elide.core.PersistentResource.loadRecordsWithSortingAndPagination(PersistentResource.java:354)
	at com.yahoo.elide.parsers.state.CollectionTerminalState.getResourceCollection(CollectionTerminalState.java:135)
	at com.yahoo.elide.parsers.state.CollectionTerminalState.handleGet(CollectionTerminalState.java:68)
	at com.yahoo.elide.parsers.state.StateContext.handleGet(StateContext.java:110)
	at com.yahoo.elide.parsers.GetVisitor.visitQuery(GetVisitor.java:34)
	at com.yahoo.elide.parsers.GetVisitor.visitQuery(GetVisitor.java:21)
	at com.yahoo.elide.generated.parsers.CoreParser$QueryContext.accept(CoreParser.java:556)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visitChildren(AbstractParseTreeVisitor.java:70)
	at com.yahoo.elide.generated.parsers.CoreBaseVisitor.visitStart(CoreBaseVisitor.java:20)
	at com.yahoo.elide.parsers.BaseVisitor.visitStart(BaseVisitor.java:47)
	at com.yahoo.elide.parsers.BaseVisitor.visitStart(BaseVisitor.java:33)
	at com.yahoo.elide.generated.parsers.CoreParser$StartContext.accept(CoreParser.java:107)
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visit(AbstractParseTreeVisitor.java:42)
	at com.yahoo.elide.Elide.get(Elide.java:369)
	at com.yahoo.elide.Elide.get(Elide.java:407)
	at com.dsbay.oms.ui.controller.elide.ElideController.jsonApiGet(ElideController.java:165)
	at com.dsbay.oms.ui.controller.elide.ElideController$$FastClassBySpringCGLIB$$c4dac9f3.invoke(<generated>)
...
```

**Comment by fabrice-etanchaud** [comment]
Hi Danuid,
I 've got the same problem while accessing a DB2i instance.

jdbc specification prohibits using first() in a forward_only resultset :

[https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSet.html#first%28%29](https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSet.html#first%28%29)

Best regards,
Fabrice

**Comment by aklish** [comment]
I have a PR that seems like it should address your issue.  Can you test with it?  It might be a few days before this gets merged to master.

**Comment by DeathByTape** [comment]
Actually, after some internal testing this looks safe. We have merged this and cut an elide 2.5.2 release with this change. We are working on releasing Elide 3.0 today, I believe though. So when you guys are ready to upgrade it should be ready :) 

Short of custom datastores, this should hopefully be a pretty painless migration.

**Comment by danuid** [comment]
@aklish @DeathByTape It worked flawless! Many thanks to both of you!!
